### ISSUE TITLE
[Bug]: Update strategy for MacOS Big Sur

### ISSUE THREAD

**Issue by tflidd** [nextcloud/desktop#6198]
### ⚠️ Before submitting, please verify the following: ⚠️

- [X] This is a **bug**, not a question or a configuration issue.
- [X] This issue is **not** already reported on Github (I've searched it).
- [X] Nextcloud Server and Desktop Client are **up to date**. See [Server Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) and [Desktop Releases](https://nextcloud.com/install/#install-clients) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/)

### Bug description

I need to state that the server and client are up to date, but putting the client up to date is the current issue.

I am on macOS big sur (11.7.10), and the Nextcloud client is version 3.2.3 and the client says 3.10.1 was available:

<img width="649" alt="Screenshot 2023-11-05 at 16 02 49" src="https://github.com/nextcloud/desktop/assets/8111200/1ae5e686-52ce-4712-acb5-9f7e782ea0c3">

But the client there is just for macOS 12+, so for me it would be the legacy version for macOS 10+, but that is version 2.6.5, so much older than the version I already have. And there seems to be an issue as well #6179.

Unfortunately, I have to stick with macOS 11, however it is still a supported version. The Client version 3.2.3 is still running, some report (see the report on the issues on 2.6.5) that even the newer version is working. Why is the version not officially supported, just a lack of testing, problems of features?

### Steps to reproduce

1. Use macOS 11.7
2. Use Nextcloud desktop client, until you hit version 3.2.3
3. It recommends to update to 3.10.1, but this version is supposed to support only macOS12+
...

### Expected behavior

More details, why some version are supported / not supported any more.
Help users with older versions, allow people to use older hardware.

### Which files are affected by this bug

nextcloud

### Operating system

Mac OS

### Which version of the operating system you are running.

macOS 11.7

### Package

Other

### Nextcloud Server version

26

### Nextcloud Desktop Client version

3.2.3

### Is this bug present after an update or on a fresh install?

Updated from a minor version (ex. 3.4.2 to 3.4.4)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### Are you using an external user-backend?

- [X] Default internal user-backend
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by marcotrevisan** [comment]
Hello, in my opinion the problematic behaviour is that the client is inviting the user to upgrade the package, but then, once upgraded, the package won't work anymore.
It sould be more "fair" to warn the user that the current MacOS version is in end of life for NC Desktop and stop any path to package upgrades.

**Comment by marcotrevisan** [comment]
In addition, I'd say that stopping support for a major OS version in a bugfix release (3.10.0 -> 3.10.1) is not very usual... I would expect such a change in a new major/feature version... my 2 cents

**Comment by philiprenich** [comment]
fwiw, the "legacy" version 2.6.5 didn't even work for me on MacOS 11.7.1 - it launches and immediately crashes, leaving only Nextcloud Extensions running in the Activity Monitor, and no toolbar icon.
I think my v3 app had auto updated itself until it no longer worked. Not sure, I wasn't expecting it to be unsupported. Going to look for a download for the v3.2.3 that supposedly works.

**Comment by philiprenich** [comment]
> fwiw, the "legacy" version 2.6.5 didn't even work for me on MacOS 11.7.1 - it launches and immediately crashes, leaving only Nextcloud Extensions running in the Activity Monitor, and no toolbar icon. I think my v3 app had auto updated itself until it no longer worked. Not sure, I wasn't expecting it to be unsupported. Going to look for a download for the v3.2.3 that supposedly works.

3.10.0 worked fine for me. (https://download.nextcloud.com/desktop/releases/Mac/Installer/)

**Comment by neonmate** [comment]
@mgallien Mac OS version 11.x is the last supported version for e.g. all MBP Late 2013. It still receives security updates (latest was https://support.apple.com/en-us/HT213915) and there is no possibility to update to Mac OS version 12.x.

Can you consider rolling back the changes made in b15369c and continue support for at least version 11.x? I understand that supporting old OS versions is painful. But sticking to an old version of the nextcloud is only a temporary solution.

**Comment by tflidd** [comment]
This was the related pull request, but it is not clear why it requires a new MacOS version:
https://github.com/nextcloud/desktop/pull/6174

**Comment by marcotrevisan** [comment]
Something is going on under the hoods: https://github.com/nextcloud/desktop/pull/6207

**Comment by aaszodi** [comment]
MacOS Big Sur 11.7 is a solid OS, it is still supported by Apple (gets regular updates). But even if it weren't, a client application **has absolutely no right to force the user to upgrade the OS**. As others have pointed out before me, the updater **must test the current OS version and refuse to install** if the new version happens to be incompatible with said OS.

Slightly off-topic but related bug: the MacOS client has an option in "Settings" that seemingly lets the user to turn off the check for updates:

![Screenshot 2023-11-25 at 17 13 26](https://github.com/nextcloud/desktop/assets/3216453/e1a17595-e0f4-45a7-9a30-3801e93307d3)

But despite the unchecked checkbox it **does check** for updates and keeps nagging the user to upgrade.

And while we are at it: I personally see **absolutely no reason to install each and every update**. Version 3.9 of the MacOS client worked for me without a hitch, just like all versions before it that I can remember (your mileage may vary of course). Yes, I gave in to the nag screen several times ... but not any more, as I am not going to upgrade from Big Sur, thank you very much.

Parting shot: Donald Knuth famously said that "premature optimization is the root of all evil". If I may humbly paraphrase him: "Unnecessary upgrades are the root of all evil". Thank you.

**Comment by xbluemonkx** [comment]
> MacOS Big Sur 11.7 is a solid OS, it is still supported by Apple (gets regular updates).

According my research November 2023 was the last security patch by apple and Big Sur has now officially reached EOL.

**Comment by Tegenwind** [comment]
Yes, Big Sur is now officially EOL, but seeing as the 3.10.0 update still supported Mojave (10.14), dropping 3 OS releases in a minor update could be considered somewhat drastic..

As for the automatic update check, this has been an issue since about June 2022 if I'm not mistaken.
There's been multiple attempts to fix it, but the behaviour remains the same.

Related issue: https://github.com/nextcloud/desktop/issues/5429

**Comment by narFnarF** [comment]
I have a Macbook Air from 2013 that I still use daily. The latest MacOS version available for it is MacOS 10.15.7 (19H2026) Catalina.

The latest version of Nextcloud sync client that works for it is version 3.10.0 (macOS "osx-19.6.0")

The client regularly prompts the user to upgrade nextcloud to 3.11 but that version won't open on macOS 10.15 Catalina.

Note: I also tried nextcloud clients 3.10.1 , 3.10.2 and 3.10.3 : They aren't compatible either.

**Comment by WolfgangFahl** [comment]
It would be great to have a legacy support version with limited features but endless compatbility ...

**Comment by sirnacnud** [comment]
Is there a reason why the minimum version was bumped up so many versions?

There are a lot of reasons to stay on older versions, in particular 10.14 was the last macOS that can run 32bit apps.

The sync client is mainly C++ code, are they planning to switch to Apple’s File Provider API?

**Comment by mannp** [comment]
I love Nextcloud, and so it's a real shame, as Nexcloud usually pushes back on big tech to stop this sort of nonsense being the norm.

OS 12 in now the minimum to sync files? Really Nextcloud? 

Perhaps skip a tiny bit of functionality to bring the masses along with your product and app rather than restricting users on apples terms. 

Syncthing supports my wifes old mac and uses the latest version, so it's possible.

**Comment by WolfgangFahl** [comment]
Please remove the tag "need triage" we are talking about dying patients here that should be saved and not be triaged as "they are dead already anways ..."

**Comment by te-online** [comment]
I just ran into this. A good first start would be to update the "legacy client" link on the website to something that actually works. Anyone knows in which repository nextcloud.com website lives? I couldn't find it! 😿

**Comment by Rello** [comment]
Hello,

we are offering a client for 10.10+ and for 12+
the legacy client is not being maintained and remains on a 3.13 version.
https://nextcloud.com/install/#desktop-files

**Comment by WolfgangFahl** [comment]
This is not a fix. The fix would be to patch the last working client not to offer updates any more.

**Comment by sirnacnud** [comment]
@Rello The 3.13.4 version doesn't work on 10.10+. I just tried running it on Catalina (10.15.7) and I get a popup saying it requires macOS 11.0 or later.

**Comment by WolfgangFahl** [comment]
It is all very frustrating. I just used https://dortania.github.io/OpenCore-Legacy-Patcher/ to upgrade one of my legacy machines to Sonoma 14.6 and even then Nextcloud latest does not work properly and crashes on start when the finder extension is enabled. You have to start things without finder extension and using a separate config to work around this startup problem. This nextlcloud software habe been a version stability night mare for me now for years. A proper LTS version with stability before features would IMHO be a very good idea.

**Comment by sirnacnud** [comment]
If you look at the diff between the last working version, `3.10.0` for pre Big Sur macOS versions, I think the issue is that the `CMAKE_OSX_DEPLOYMENT_TARGET` is set to `12.0` so it would enable `BUILD_FILE_PROVIDER_MODULE`.

Full diff here: https://github.com/nextcloud/desktop/compare/v3.10.0...v3.13.4
